src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -124,9 +136,18 @@ private function doProcessValue($value, $isRoot = false)
      *
      * @return array
      */
-    private function autowireCalls(\ReflectionClass $reflectionClass, array $methodCalls)
+    private function autowireCalls(\ReflectionClass $reflectionClass, array $methodCalls, bool $isRoot): array
     {
-        foreach ($methodCalls as $i => $call) {
+        $this->methodCalls = $methodCalls;
+        if ($isRoot && $this->container->has($this->decoratedId = $this->currentId.'.inner')) {
```

Comment:
The inner service name can be custom see `Definition::getDecoratedService()` and https://symfony.com/doc/current/service_container/service_decoration.html

Reply:
Indeed, but the renamed id is discarded here https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php#L41

We may introduce a new `originalRenameId` in `Definition` to keep it, but considering that autowiring is all about no-config, does it worth it? I mean, probably nobody will use both `autowire: true` and `decoration_inner_name` together (we can even throw in this case). WDYT?

Reply:
decoration implies manual service definition (`decorates: ...`) so I think it makes sense to support `decoration_inner_name`, and throwing now would be a BC break. If we can't support it easily then better let autowiring fails I think, error messages are useful enough